src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -505,12 +505,57 @@ private void drawHUD(Graphics g, boolean includeTileDetails)
       overlayIsLeft = false;
     }
 
+    // Draw the Commander's image, current funds, and power level.
     CommanderOverlayArtist.drawCommanderOverlay(g, myGame.activeCO, overlayIsLeft);
 
+    drawTurnCounter(g, overlayIsLeft);
+
+    // Draw terrain defense and unit status.
     if( includeTileDetails )
       MapTileDetailsArtist.drawTileDetails(g, myGame.activeCO.myView, myGame.getCursorCoord(), overlayIsLeft);
   }
 
+  private int lastTurnNum = -1;
+  private BufferedImage turnNumImage;
+  private void drawTurnCounter(Graphics g, boolean counterIsRight)
+  {
+    // Generate the turn-counter image.
+    int turnNum = myGame.getCurrentTurn();
+    if( lastTurnNum != turnNum )
+    {
+      System.out.println("rebuilding turn image.");
+      lastTurnNum = turnNum;
+      BufferedImage day = SpriteUIUtils.getTextAsImage("Turn ");
+      BufferedImage dayNum = SpriteUIUtils.getBoldTextAsImage(Integer.toString(turnNum));
+      int width = day.getWidth() + dayNum.getWidth();
+      int height = dayNum.getHeight();
+
+      turnNumImage = SpriteLibrary.createTransparentSprite(width, height);
+      Graphics dcg = turnNumImage.getGraphics();
+      dcg.drawImage(day, 0, 0, null);
+      dcg.drawImage(dayNum, turnNumImage.getWidth()-dayNum.getWidth(), 0, null);
+    }
+
+    // Draw the turn counter.
+    int xDraw = (counterIsRight
+        ? (SpriteOptions.getScreenDimensions().width / SpriteOptions.getDrawScale()) - 2 - turnNumImage.getWidth()
+        : 2);
+    int yDraw = 3;
+
+    // Draw a CO-colored background with the counter.
+    int arcW = turnNumImage.getHeight()+4;
+    g.setColor(UIUtils.getMapUnitColors(myGame.activeCO.myColor).paletteColors[4]); // 0 is darker, 5 is lighter.
```

Comment:
Is it intentional to make this a different shade than the CO panel? Pretty sure that one just uses the "signature"/key color

Reply:
Yes. The CO panel is a sprite, not procedurally-generated. But also no, since it looks like I ended up selecting the representative color anyway. I can change it back to simplify.